### DISCUSSION TITLE
MOSAIC can not run NS-3 federate in the docker container

### DISCUSSION THREAD

**Discussion by DFCui** [eclipse/mosaic#149] [Q&A] [answered]
Hello all,
I built the NS-3 federate docker image on windows following the documentation of MOSAIC. Here is the link. [https://www.eclipse.org/mosaic/docs/simulators/network_simulator_ns3/]. As stated in the documentation, I added the image name to the runtime.json, and enable NS-3 federate in Tiergarten scenario. When I run the MOSAIC with Tiergarten scenario, the NS-3 federate starts in the docker container, however, MOSAIC is not able to connect with the NS-3 federate and it stops at starting NS-3 federate. 
Is there anyone having the same issue? Could you please provide some suggestions about how to solve it? Thank you very much.

**Comment by kschrab** [comment, marked as answer]
Try to add `-Dmosaic.docker.no-detach=true` to the java call in the `mosaic.bat` or `mosaic.sh`, e.g.

```
java -Dmosaic.docker.no-detach=true -Xmx%javaMemorySizeXmx% %javaRemoteDebugging% -cp !libs! org.eclipse.mosaic.starter.MosaicStarter %*
```
